### PR TITLE
Weapon changes

### PR DESCRIPTION
**Pull request by haxtonsale**
* Renamed `eWeaponsRarity` to `WeaponRarity`
* Made small changes to `PickupWeapon`
* Changed pistol's item definition index to 209
* If player picks up a weapon and he has it (or a reskin) equipped in their loadout, `TF2_CreateAndEquipWeapon` now gives the loadout item instead.

### CODE REVIEW

File: addons/sourcemod/scripting/include/superzombiefortress.inc [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #endif
 #define _superzombiefortress_included
 
-enum eWeaponsRarity
+enum WeaponRarity
```
Comment (FortyTwoFortyTwo) [inline addons/sourcemod/scripting/include/superzombiefortress.inc:6]: May well then also rename `eWeaponsRarity_Common` to `WeaponsRarity_Common` etc while were breaking sub-plugins with `eWeaponsRarity`

File: addons/sourcemod/scripting/szf/config.sp [hunk 2]
```diff
@@ -1,5 +1,6 @@
-#define CONFIG_WEAPONS "configs/szf/weapons.cfg"
-#define CONFIG_CLASSES "configs/szf/classes.cfg"
+#define CONFIG_WEAPONS       "configs/szf/weapons.cfg"
+#define CONFIG_CLASSES       "configs/szf/classes.cfg"
+#define CONFIG_WEAPONINDEXES "configs/szf/weaponindexes.cfg"
```
Comment (FortyTwoFortyTwo) [inline addons/sourcemod/scripting/szf/config.sp:3]: `reskins.cfg` or `weaponreskins.cfg` may be a better name when it only used to match reskins with original.

File: addons/sourcemod/scripting/szf/config.sp [hunk 3]
```diff
@@ -434,6 +437,38 @@ ArrayList Config_LoadInfectedClasses()
...
+			kv.GetSectionName(sName, sizeof(sName));
+			int iOrigIndex = StringToInt(sName);
+			
+			char sValue[512];
+			kv.GetString(NULL_STRING, sValue, sizeof(sValue));
+			
+			char sValueExploded[64][8];
+			int iCount = ExplodeString(sValue, " ", sValueExploded, sizeof(sValueExploded), sizeof(sValueExploded[]));
+			
+			if (sValueExploded[0][0] != '\0')
```
Comment (FortyTwoFortyTwo) [inline addons/sourcemod/scripting/szf/config.sp:461]: either just check `iCount` is not 0 or don't bother checking and let `for` skip it

File: addons/sourcemod/scripting/szf/stocks.sp [hunk 4]
```diff
@@ -742,6 +759,19 @@ stock void CheckClientWeapons(int iClient)
 	}
 }
 
+stock int GetOriginalItemDefIndex(int iIndex)
+{
+	int iOrigIndex;
+	
+	char sIndex[8];
```
Comment (FortyTwoFortyTwo) [inline addons/sourcemod/scripting/szf/stocks.sp:766]: 12 char length is min needed to fit for 32-bit int, though we'll never get that high for item def index

File: addons/sourcemod/scripting/szf/pickupweapons.sp [hunk 5]
```diff
@@ -268,10 +266,11 @@ void PickupWeapon(int iClient, Weapon wep, int iTarget)
 	g_bCanPickup[iClient] = false;
 	CreateTimer(PICKUP_COOLDOWN, Timer_ResetPickup, iClient);
 	
+	//Weapon pickup quote
 	char sSound[PLATFORM_MAX_PATH];
-	TFClassType nClass = TF2_GetPlayerClass(iClient);
+	TFClassType iClass = TF2_GetPlayerClass(iClient);
```
Comment (FortyTwoFortyTwo) [inline addons/sourcemod/scripting/szf/pickupweapons.sp:271]: I use `n` notation when it an integer, but not pure `int`, but meh

File: addons/sourcemod/scripting/szf/config.sp [hunk 6]
```diff
@@ -12,10 +13,12 @@ enum struct ConfigMelee
 
 ConfigMelee g_ConfigMeleeDefault;
 ArrayList g_aConfigMelee;
+StringMap g_OrigWeaponIndexes;
 
 void Config_Init()
 {
 	g_aConfigMelee = new ArrayList(sizeof(ConfigMelee));
+	g_OrigWeaponIndexes = Config_LoadWeaponIndexes();
```
Comment (FortyTwoFortyTwo) [inline addons/sourcemod/scripting/szf/config.sp:21]: Should be done in `Config_Refresh` to allow map-change or `sm_szfreload` refresh it.

File: addons/sourcemod/gamedata/szf.txt [hunk 7]
```diff
@@ -46,6 +52,11 @@
 				"linux"		"482"
 				"windows"	"475"
 			}
+			"CBaseEntity::GetBaseEntity"
+			{
+				"windows" "5"
+				"linux" "6"
```
Comment (FortyTwoFortyTwo) [inline addons/sourcemod/gamedata/szf.txt:58]: tab

**Comment by haxtonsale** [comment]
Done

File: addons/sourcemod/configs/szf/reskins.cfg [hunk 8]
```diff
@@ -0,0 +1,102 @@
+"Weapon Indexes"
```
Comment (FortyTwoFortyTwo) [inline addons/sourcemod/configs/szf/reskins.cfg:1]: Should also be renamed to `Reskins`

**Pull request merged**